### ISSUE TITLE
csdebug.h missing when installing on Debian Linux Jessie

### ISSUE THREAD

**Issue by itree** [nwhetsell/csound-api#2]
When installing on debian linux Jessie

get the following error.

csound-api@0.6.6 install /home/csound/node_modules/csound-apinode-gyp rebuild

make: Entering directory '/home/csound/node_modules/csound-api/build'
  CXX(target) Release/obj.target/csound-api/src/csound-api.o
../src/csound-api.cpp:2:21: fatal error: csdebug.h: No such file or directory
 #include <csdebug.h>
                     ^
compilation terminated.
csound-api.target.mk:92: recipe for target 'Release/obj.target/csound-api/src/csound-api.o' failed
make: **\* [Release/obj.target/csound-api/src/csound-api.o] Error 1
make: Leaving directory '/home/csound/node_modules/csound-api/build'

**Comment by nwhetsell** [comment]
Thanks for reporting this. This exposed several issues:
1. The GYP file [(binding.gyp)](https://github.com/nwhetsell/csound-api/blob/master/binding.gyp) was just not set up for building on Linux.
2. Updating the GYP file took care of finding csdebug.h, but building still failed because [jessie seems to use Csound 6.03.2](https://packages.debian.org/jessie/libcsound64-dev), and Csound’s debugger functions and structures changed a bit in Csound 6.04. For now, my fix for this is to require Csound 6.04 or later to use debugger functions.
3. Because I missed a [handle scope](https://github.com/nwhetsell/csound-api/blob/6d49c550441ee16264cec099e3a8e1b66052da1a/src/csound-api.cpp#L35), the tests of using JavaScript functions as Csound callbacks failed in recent versions of Node.js.

I can now compile `csound-api` on jessie. Before I publish an update to npm, can you try to build `csound-api`? I think you can do this by running:

``` sh
git clone https://github.com/nwhetsell/csound-api csound-api
cd csound-api
npm install nan
node-gyp rebuild
```

Also, can you look at the [Linux installation instructions](https://github.com/nwhetsell/csound-api#on-linux), and let me know if they make sense? One thing I’m unsure about is using `sudo` to install Boost and Csound.

**Comment by nwhetsell** [comment]
I’ve gone ahead and published an updated package to npm that I believe will install on Linux, but please feel free to reopen this issue if you have any trouble.
